src/components/PolygonEditor.tsx

```diff
@@ -77,24 +90,31 @@ export const PolygonEditor = ({ value, onChange, context }: PolygonEditorProps)
               <HorizontalGroup>
                 <div>
                   <Label>Name</Label>
-                  <Input {...register(`areas.${index}.name` as const)} defaultValue={field.name} placeholder='Enter the area name' />
+                  <Input
+                    {...register(`areas.${index}.name` as const)}
+                    defaultValue={field.name}
+                    placeholder="Enter the area name"
+                  />
                 </div>
                 <div>
                   <Label>Color</Label>
-                  <Input
-                    prefix={
-                      <ColorPicker
-                        color={watch(`areas.${index}.color`) || 'red'}
-                        onChange={(color) => {
-                          setValue(`areas.${index}.color`, color);
-                        }}
-                      />
-                    }
-                    {...register(`areas.${index}.color` as const)}
+                  <ColorPickerInput
+                    value={watch(`areas.${index}.color`)}
                     defaultValue={field.color}
+                    {...register(`areas.${index}.color` as const)}
+                    onChange={(color) => {
+                      setValue(`areas.${index}.color`, color);
+                    }}
```

Comment:
I changed it to a color input because the previous solution was not compatible with the polygon creator."